### PR TITLE
Document `tools.web` package

### PR DESCRIPTION
**Pull request by dgw**
Draft because it's based on the #1616 branch, which isn't yet merged.

See 6d53ced for the relevant changes. I'll keep an eye on this after merging its parent.

### CODE REVIEW

**Comment by dgw** [comment]
Poked it with a rebase just to be safe; GitHub appeared not to figure out the merge on its own.

File: sopel/tools/web.py [hunk 1]
```diff
@@ -62,12 +114,24 @@ def entity(match):
...
 
 # Identical to urllib2.quote
 def quote(string, safe='/'):
-    """Like urllib2.quote but handles unicode properly."""
+    """Safely encodes a string for use in a URL.
+
+    :param str string: the string to quote
+    :param str safe: a list of characters that should not be quoted
+    :return str: the ``string`` with special characters URL-encoded
+
```
Comment (Exirel) [inline sopel/tools/web.py]:
```suggestion
```

File: sopel/tools/web.py [hunk 2]
```diff
@@ -62,12 +114,24 @@ def entity(match):
 
 
 def decode(html):
+    """Decode HTML entities into Unicode text.
+
+    :param str html: the HTML page or snippet to process
+    :return str: ``html`` with all entity references replaced
+
```
Comment (Exirel) [inline sopel/tools/web.py]:
```suggestion
```

File: sopel/tools/web.py [hunk 3]
```diff
@@ -44,13 +47,62 @@
...
 
 def entity(match):
+    """Convert an entity reference to the appropriate character.
+
+    :param str match: the entity name or code, as matched by
+        :py:const:`r_entity`
+    :return str: the Unicode character corresponding to the given ``match``
+        string, or a fallback representation if the reference cannot be
+        resolved to a character
+
```
Comment (Exirel) [inline sopel/tools/web.py]:
```suggestion
```

**Comment by dgw** [comment]
Nitpicks addressed. I'll consider this ready to merge now, barring any further nitpicks. :rocket:

**Pull request merged**